File: code/31190.pas
program Bai190;
  var a,b, c,d, TS, MS, x, y : integer;
  function UCLN(m,n: integer): integer;
  begin
    while m <> n  do if m > n then m:= m - n else n := n - m;
    UCLN := m;
  end;
  begin
  readln(a,b);
  readln(c,d);
  TS := a*d + b*c;
  MS := b*d;
  x := TS div UCLN(TS,MS);
  y := MS div UCLN(TS,MS);
  writeln (x, '/', y);
end.

File: code/31190.py
a, b = [int(x) for x in input().split()]
c, d = [int(x) for x in input().split()]
def UCLN(x, y):
  while x != y:
    if x > y :
     x = x - y 
    else:
     y = y - x
  return(x)
T = a*d + b*c
M = b*d
print((T// UCLN(T,M)),M // UCLN(T, M))